### ISSUE TITLE
Add remaining states to NavBar component

### ISSUE THREAD

**Issue by GBKS** [saving-satoshi/saving-satoshi#66]
The NavBar that is shown during lessons is missing various details ([Figma](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=161%3A1652&t=nxCE26gS2fD24CVh-1)):
- It has a totally different mobile layout with an expandable challenge menu
- The top line of text is incorrectly formatted ("chapter-1" should be "Chapter 2")
- The chapter name is missing on chapter intro pages ([Example](https://savingsatoshi.com/chapters/chapter-1))
- It's missing help and account buttons
- Challenge numbers are missing states (locked, current, completed)
- Back button should go linearly back in the learning experience and not do a browser back

Design:
![image](https://user-images.githubusercontent.com/695901/211275851-eff34b7f-2d5f-4240-9000-1ffac12f1275.png)

Screenshot:
![image](https://user-images.githubusercontent.com/695901/211276329-e789bfce-7699-453b-b5f9-84f9692c7c21.png)

**Comment by ecurrencyhodler** [comment]
Kevin will be taking a look at this.

**Comment by GBKS** [comment]
Awesome, so an amazing thing would be to have the expand/collapse fully animated on mobile, where the nav items slide in in a staggered fashion, and the hamburger icon nicely transitions to an x and back. Here's a mockup:

![image](https://user-images.githubusercontent.com/695901/214518522-87c1d3fc-943e-4a66-b5ca-7e097be32c7f.png)

I implemented something fairly similar on my own website [here](https://www.germanysbestkeptsecret.com/). I know it's pretty intricate to execute, so let me know what your thoughts are.

**Comment by ecurrencyhodler** [comment]
K change of plans. Kevin is focusing on this issue https://github.com/orgs/saving-satoshi/projects/2?pane=issue&itemId=18397150

Sashwat said he will take a look at this one potentially early next week instead if he's able to finish up his studies by EoW.

**Comment by GBKS** [comment]
Sounds good. @shaavan, before you start this, let's chat and talk through the component first, since it's fairly intricate.

**Comment by shaavan** [comment]
Sure! @GBKS 

Let's discuss about this on Slack!

**Comment by ecurrencyhodler** [comment]
Resolved by https://github.com/saving-satoshi/saving-satoshi/pull/140.

**Comment by GBKS** [comment]
This is not closed, as #140 only addresses a small part of the issue.

**Comment by ecurrencyhodler** [comment]
Woops. Thanks for reopening.

**Comment by ecurrencyhodler** [comment]
Following up on this issue. Remaining items are:

- [ ]  It has a totally different mobile layout with an expandable challenge menu
- [ ] The top line of text is incorrectly formatted ("chapter-1" should be "Chapter 2")
- [ ] - The chapter name is missing on chapter intro pages ([Example](https://savingsatoshi.com/chapters/chapter-1))
- [ ] - It's missing ~help and~ account buttons
- [ ] - Challenge numbers are missing states (locked, current, completed)
- [ ] - Back button should go linearly back in the learning experience and not do a browser back 

The first item of a Nav bar for mobile is in progress with https://github.com/saving-satoshi/saving-satoshi/pull/146

Second item the topline text for ch. 2 is current inaccessible as it is blocked by locked chapters. Leaving it unmarked so we know to circle back on it but I don't believe this is a blocker for the demo.

Third item remains open.

Fourth item, I removed requirement for help button based on [this discussion.](https://github.com/saving-satoshi/saving-satoshi/pull/146#issuecomment-1427494966). But the account button I believe has been resolved.

Fifth item has been resolved as they now reflect states.

Sixth item is currently being discussed and scoped [in slack](https://bitcoindesign.slack.com/archives/C0442BRGJ5U/p1676630421066219) with regards to design.

**Comment by gadabra** [comment]
[ecurrencyhodler](https://github.com/ecurrencyhodler) Thank you for conversation

This is my other account. You can assign on me.

**Comment by ecurrencyhodler** [comment]
Okay I think the 2nd item "The top line of text is incorrectly formatted ("chapter-1" should be "Chapter 2")" has actually been resolved when I examine the screenshots provided in the initial issue and prod.

Here is the current state of prod:
<img width="1096" alt="Screenshot 2023-02-20 at 12 21 31 PM" src="https://user-images.githubusercontent.com/32662508/220192165-2641395e-a669-4678-8b48-9a1d658c461b.png">

**Comment by GBKS** [comment]
@ecurrencyhodler sorry for causing confusion around the "arrow left" button in the nav, I think I mixed something up. So if that button simply goes backward in the lesson series, then the user has no way of quickly exiting the learning experience (we have not other logo or home button anywhere). So my proposal would be to repurpose it to a home button that leads to the chapters overview page. The user can still go backwards and forwards in the lesson content using the browser UI back/forwards, and the numerical lesson buttons in the nav bar. This would affect current work in #203.

**Comment by ecurrencyhodler** [comment]
Thanks christoph!

That means the 3rd and 6th issue have been resolved. Only pending item is the first one which shashwat is close to resolving. I'm going to create a separate issue to create a new home button that redirects to the chapters overview page.

**Comment by ecurrencyhodler** [comment]
Closing this issue as shaavan's pr has been merged. Great work everyone! This was a massive one.
